runtime/vm/resolvesupport.cpp

```diff
@@ -1979,6 +2086,62 @@ resolveMethodHandleRef(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpInde
 
 }
 
+j9object_t
+resolveMethodHandle(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpIndex, UDATA resolveFlags)
+{
+	j9object_t memberName = NULL;
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+	J9JavaVM *vm = vmThread->javaVM;
+	bool canRunJavaCode = J9_ARE_NO_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_JIT_COMPILE_TIME | J9_RESOLVE_FLAG_REDEFINE_CLASS);
+	J9RAMMethodRef *ramCPEntry = (J9RAMMethodRef*)ramCP + cpIndex;
+	UDATA invokeCacheIndex = ramCPEntry->methodIndexAndArgCount >> 8;
+	J9Class *ramClass = J9_CLASS_FROM_CP(ramCP);
+	J9InvokeCacheEntry *resultEntry = ((J9InvokeCacheEntry *)ramClass->invokeCache) + invokeCacheIndex;
+	memberName = resultEntry->target;
+
+	if (NULL != memberName) {
+		return memberName;
+	} else if (!canRunJavaCode) {
+		return NULL;
+	}
+
+	J9ROMMethodRef *romMethodRef = (J9ROMMethodRef *)&ramCP->romConstantPool[cpIndex];
+	J9ROMNameAndSignature *nameAndSig = J9ROMMETHODREF_NAMEANDSIGNATURE(romMethodRef);
+
+	/* Resolve the class. */
+	J9Class *resolvedClass = resolveClassRef(vmThread, ramCP, romMethodRef->classRefCPIndex, resolveFlags);
+	if (resolvedClass != NULL) {
+		/* Create appendix array*/
+		J9Class *objectClass = J9VMJAVALANGOBJECT(vm);
+		J9Class *arrayClass = objectClass->arrayClass;
+		if (NULL == arrayClass) {
+			/* Allocate an array class */
+			J9ROMArrayClass* arrayOfObjectsROMClass = (J9ROMArrayClass*)J9ROMIMAGEHEADER_FIRSTCLASS(vmThread->javaVM->arrayROMClasses);
+			arrayClass = internalCreateArrayClass(vmThread, arrayOfObjectsROMClass, objectClass);
+		}
+		j9object_t appendix = vm->memoryManagerFunctions->J9AllocateIndexableObject(vmThread, arrayClass, 1, J9_GC_ALLOCATE_OBJECT_INSTRUMENTABLE);
+
+		sendResolveOpenJDKMethodHandle(vmThread, ramCP, cpIndex, MH_REF_INVOKEVIRTUAL, resolvedClass, nameAndSig, appendix);
```

Comment:
Where is the java method this calls?  I don't see it in this PR which is making it hard to review.

I'm wondering why the object array is being allocated here in native code rather than in the java code.  Being able to see the java method being called would make it clear if that's an option.  Always better to do the allocations in java then in the VM native code.

Additionally, I'm wondering why we're discarding the array and storing the data into a native structure.  The resolution code would be simpler if we stored the array rather the discarding it.  It would mean indexing to get the array and then derefing thru it

Reply:
The java method called in `sendResolveOpenJDKMethodHandle` is `MethodhandleHelper.resolveMethodHandle()` which is part of #10830 commit : https://github.com/eclipse/openj9/pull/10830/commits/628621af13063ff19389e32968006a10049e2edc

The OpenJDK API will return 2 values during resolution, 1 through the return value and 1 through the single element array passed in as argument.

Reply:
In which PR is the change to the bytecode interpreter to call the MH?

(Noting here so I don't forget - the appendix[0] value can also be null)